sources/bloodhound/src/lib.rs

```diff
@@ -1,3 +1,123 @@
+use std::io::{self, BufRead, BufReader};
+use std::{ffi::OsStr, fs::File, process::Command};
+
 pub mod args;
 pub mod output;
 pub mod results;
+
+/// Reads a file and checks if the given `search_str` is present in its contents.
+pub fn look_for_string_in_file(path: &str, search_str: &str) -> Result<bool, io::Error> {
+    let reader = BufReader::new(File::open(path)?);
+    Ok(reader
+        .lines()
+        .any(|line| line.unwrap_or_default().contains(search_str)))
+}
+
+/// Executes a command and checks if the give `search_str` is in the output.
+pub fn look_for_string_in_output<I, S>(cmd: &str, args: I, search_str: &str) -> Option<bool>
```

Comment:
Wouldn't it be simpler to execute the command in `execute`, and let each checker decide how to treat non-zero exit codes (I'm assuming there will be commands that exit non-zero, but that's expected for some checkers)?. Otherwise, in here the function considers non-zero exit codes as “couldn't find the string” but there might be cases where the string will be found on the stderr.

Also, elsewhere there is a function to check for many strings, instead of just one, I think you can get rid of this function and just use the other with the one string passed to it.

Reply:
This is really a convenience to library consumers to have a simple interface to call. I've updated the function though to just delegate to the `_strings_` function.

As far as non-zero exit codes, that shouldn't be a need for other checkers. The commands should be expected to run, it's the `Option<bool>` that is of interest. So if a check needs to make sure a string value does _not_ exist, then it can check that bool result as appropriate.

If a check does end up needing some odd handling (which I haven't seen yet), then it can duplicate this code as a worst case scenario. But I definitely would not want to repeat this code in every `execute` block if it's just repetition.